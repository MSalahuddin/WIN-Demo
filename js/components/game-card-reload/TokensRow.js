import React from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import {Platform} from 'react-native';
import Text, { FONT_FAMILY, SIZE } from '../common/Text';
import { coin } from '../../../assets/images';
import { scaleHeight, scale } from '../../platformUtils';
import { gameCardReloadStrings } from '../../stringConstants';
import { color } from '../../styles';
import { convertNumberToStringWithComma, capitalize } from '../../utils';

const Container = styled.View`
  height: ${scaleHeight(88)};
  background-color: ${color.grayWhite};
`;

const RowContainer = styled.View`
  flex-direction: row;
  justify-content: space-around;
`;

const TokenTextContainer = styled.View`
  align-items: center;
  margin-bottom: ${scaleHeight(8)};
  margin-top: ${scaleHeight(22)};
`;

const TokenSection = styled.View`
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const IconTextRow = styled.View`
  flex-direction: row;
  justify-content: center;
  align-items: flex-start;
  margin-top: ${scaleHeight(5)};
`;

const StyledImage = styled.Image`
  width: ${scale(18)};
  height: ${scale(18)};
  margin-right: ${scale(5)};
`;

const StyledText = styled(Text)`
  margin-top: ${scaleHeight(Platform.OS==='android' ? 0 : 3)};
`;

const TokensRow = ({ oneTimeTokenAmount, monthlyTokenAmount, name }) => {
  const sectionData = [
    { title: gameCardReloadStrings.tokensToday, token: convertNumberToStringWithComma(oneTimeTokenAmount) },
    {
      title: gameCardReloadStrings.monthlyTokens,
      token: convertNumberToStringWithComma(monthlyTokenAmount)
    }
  ];
  return (
    <Container>
      <TokenTextContainer>
        <Text color={color.silver} fontFamily={FONT_FAMILY.REGULAR} size={SIZE.XXSMALL}>
          {`${capitalize(name)} ${gameCardReloadStrings.subscriptionPack}`}
        </Text>
      </TokenTextContainer>
      <RowContainer>
        {sectionData.map(({ title, token }, index) => (
          <TokenSection key={index}>
            <Text size={SIZE.XXSMALL}>{title}</Text>
            <IconTextRow>
              <StyledImage source={coin} resizeMode="contain" />
              <StyledText size={SIZE.XSMALL} fontFamily={FONT_FAMILY.REGULAR} color={color.lightGrey}>
                {token}
              </StyledText>
            </IconTextRow>
          </TokenSection>
        ))}
      </RowContainer>
    </Container>
  );
};

TokensRow.propTypes = {
  monthlyTokenAmount: PropTypes.number.isRequired,
  oneTimeTokenAmount: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired
};

export default TokensRow;
