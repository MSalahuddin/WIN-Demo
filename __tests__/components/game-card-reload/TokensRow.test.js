import React from 'react';
import { render } from 'react-native-testing-library';

import TokensRow from '../../../js/components/game-card-reload/TokensRow';

describe('Component TokensRow', () => {
  test('renders correctly', () => {
    const output = render(<TokensRow monthlyTokenAmount={1} oneTimeTokenAmount={1} name="" />);
    expect(output).toMatchSnapshot();
  });
});
