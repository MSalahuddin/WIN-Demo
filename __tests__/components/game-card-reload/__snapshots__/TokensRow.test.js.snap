// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TokensRow renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F9F9F9",
        "height": 176,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "marginBottom": 16,
          "marginTop": 44,
        },
      ]
    }
  >
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#8B8B8B"
      fontFamily="REGULAR"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#8B8B8B",
            "fontFamily": "CalibreRegular",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#8B8B8B",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
    >
       Subscription Pack
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        alignCenter={false}
        allowFontScaling={true}
        color="#000000"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="XXSMALL"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Calibre-Medium",
              "fontSize": 21,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#000000",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
          ]
        }
      >
        Tokens Today
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 10,
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 27,
                "marginRight": 7.5,
                "width": 27,
              },
            ]
          }
        />
        <Text
          alignCenter={false}
          allowFontScaling={true}
          color="#B7B7B7"
          fontFamily="REGULAR"
          isUnderlined={false}
          isUppercase={false}
          maxFontSizeMultiplier={1.3}
          size="XSMALL"
          style={
            Array [
              Object {
                "color": "#B7B7B7",
                "fontFamily": "CalibreRegular",
                "fontSize": 24,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#B7B7B7",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "none",
              },
              Object {
                "marginTop": 6,
              },
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        alignCenter={false}
        allowFontScaling={true}
        color="#000000"
        fontFamily="MEDIUM"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        size="XXSMALL"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Calibre-Medium",
              "fontSize": 21,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#000000",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
          ]
        }
      >
        Monthly Tokens
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 10,
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 27,
                "marginRight": 7.5,
                "width": 27,
              },
            ]
          }
        />
        <Text
          alignCenter={false}
          allowFontScaling={true}
          color="#B7B7B7"
          fontFamily="REGULAR"
          isUnderlined={false}
          isUppercase={false}
          maxFontSizeMultiplier={1.3}
          size="XSMALL"
          style={
            Array [
              Object {
                "color": "#B7B7B7",
                "fontFamily": "CalibreRegular",
                "fontSize": 24,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#B7B7B7",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "none",
              },
              Object {
                "marginTop": 6,
              },
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
  </View>
</View>
`;
